server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/cassandra/CassandraTaskExecutionDetailsProjectionDAOTest.java

```diff
@@ -111,4 +120,65 @@ void listDetailsShouldReturnLastUpdatedRecords() {
         Stream<TaskExecutionDetails> taskExecutionDetails = testee.listDetails().toStream();
         assertThat(taskExecutionDetails).containsOnly(TASK_EXECUTION_DETAILS_UPDATED());
     }
+
+    @Test
+    void listBeforeDateShouldReturnCorrectEntry() {
+        TaskExecutionDetails taskExecutionDetails1 = new TaskExecutionDetails(TASK_ID(),
+            TaskType.of("type"),
+            TaskManager.Status.COMPLETED,
+            ZonedDateTime.ofInstant(Instant.parse("2000-01-01T00:00:00Z"), ZoneId.systemDefault()),
+            TaskExecutionDetailsFixture.SUBMITTED_NODE(),
+            Optional::empty,
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+
+        TaskExecutionDetails taskExecutionDetails2 = new TaskExecutionDetails(TASK_ID_2(),
+            TaskType.of("type"),
+            TaskManager.Status.COMPLETED,
+            ZonedDateTime.ofInstant(Instant.parse("2000-01-20T00:00:00Z"), ZoneId.systemDefault()),
+            TaskExecutionDetailsFixture.SUBMITTED_NODE(),
+            Optional::empty,
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty(),
+            Optional.empty());
+
+        testee.saveDetails(taskExecutionDetails1).block();
+        testee.saveDetails(taskExecutionDetails2).block();
+
+        assertThat(Flux.from(testee.listDetailsByBeforeDate(Instant.parse("2000-01-15T12:00:55Z"))).collectList().block())
+            .containsOnly(taskExecutionDetails1);
+    }
+
+    @Test
+    void removeShouldDeleteAssignEntry() {
+        TaskExecutionDetails taskExecutionDetails1 = new TaskExecutionDetails(TASK_ID(),
+            TaskType.of("type"),
+            TaskManager.Status.COMPLETED,
```

Comment:
Q: What if we remove an in-progress task?

Reply:
Not sure, 
But IMO it will not happen in real world, this endpoint for remove "old task" with "olderThan" parameter. 
Eg: I don't think has a task still in-progress in more than 15 days. And admin try to remove tasks too early.

Of course, here is code logic, let wait other comment

Reply:
Re-index task/ Ham report with million messages could take that long IMO